### ISSUE TITLE
AL Round-Trip Test Plan

### ISSUE THREAD

**Issue by JohnNKing** [CDCgov/trusted-intermediary#965]
# Story
As a PHL or HCO, so that we're confident the various branches of our workflows will work under ETOR, test various scenarios that use test orders and corresponding results from partner systems (e.g. EHR, LIMS).

## Pre-conditions
- Data gap analysis efforts have been completed
- HCO and PHL can send an receive orders and results messages to each other via ReportStream
- Connectivity and integration testing have been completed for both HCO and PHL

## Acceptance Criteria
- [ ] We've received approvals of the test scenarios from our HCO and PHL partners

## Tasks
- [x] Refine the mnemonics list
- [x] Test plan refined based on data gap work
- [x] Request feedback on plan from AL PHL
- [x] Request feedback on plan from Baptist
- [x] Test scenarios refined based on data gap work
- [x] Request feedback on scenarios from AL PHL
- [x] Request feedback on scenarios from Baptist
- [ ] Update test scenarios based on partner feedback

## Definition of Done
- [ ] Documentation tasks completed
  - [ ] Documentation and diagrams created or updated
    - [ ] Implementation guide (`/ig` folder)
    - [ ] ADRs (`/adr` folder)
    - [ ] Main `README.md`
    - [ ] Other READMEs in the repo
    - [ ] If applicable, update the `ReportStream Setup` section in `README.md`
  - [ ] [Threat model](https://lucid.app/lucidchart/8c6e8d37-2612-42a8-ac57-150c83f8e29e/edit) updated
  - [ ] API documentation updated
- [ ] Code quality tasks completed
  - [ ] Code refactored for clarity and no design/technical debt
  - [ ] Adhere to separation of concerns; code is not tightly coupled, especially to 3rd party dependencies
  - [ ] Code is reviewed or developed by pair; 1 approval is needed but consider requiring an outside-the-pair reviewer
  - [ ] Code quality checks passed
- [ ] Security & Privacy tasks completed
  - [ ] Security & privacy gates passed
- [ ] Testing tasks completed
  - [ ] Load tests passed
  - [ ] Unit test coverage of our code >= 90%
- [ ] Build & Deploy tasks completed
  - [ ] Build process updated
  - [ ] API(s) are versioned
  - [ ] Feature toggles created and/or deleted.  Document the feature toggle
  - [ ] Source code is merged to the main branch

## Research Questions
- *Optional: Any initial questions for research*

## Decisions
- *Optional: Any decisions we've made while working on this story*

## Notes
- *Optional: Any reference material or thoughts we may need for later reference*

**Comment by JohnNKing** [comment]
Pending test scenario work based on profiles.

We may want to split off test execution working into separate story.

**Comment by JohnNKing** [comment]
@JohnNKing -- Need to split this as with the CA Round Trip Planning vs execution

**Comment by sfradkin** [comment]
need feedback from AL lab.  however, we may put this on temporary hold as we're currently working on the first functional round trip test.

**Comment by sfradkin** [comment]
waiting on out of range values for specific test cases.

**Comment by JScott-JMC** [comment]
Knowledge transfer is complete for handover of the AL engagement to the DETOR team.  This story can be closed.

@JohnNKing
